pkg/apis/config/scheme/scheme_test.go

```diff
@@ -0,0 +1,336 @@
+		data         []byte
+		wantErr      string
+		wantProfiles []defaultconfig.KubeSchedulerProfile
+	}{
+		//v1beta1 tests
+		{
+			name: "v1beta1 all plugin args in default profile",
+			data: []byte(`
+apiVersion: kubescheduler.config.k8s.io/v1beta1
+kind: KubeSchedulerConfiguration
+profiles:
+- schedulerName: scheduler-plugins
+  pluginConfig:
+  - name: Coscheduling
+    args:
+      permitWaitingTimeSeconds: 10
+      deniedPGExpirationTimeSeconds: 3
+  - name: NodeResourcesAllocatable
+    args:
+      mode: Least
+      resources:
+      - name: cpu
+        weight: 1000000
+      - name: memory
+        weight: 1
+  - name: TargetLoadPacking
+    args:
+      targetUtilization: 60 
+      defaultRequests:
+        cpu: "1000m"
+      defaultRequestsMultiplier: "1.8"
+      watcherAddress: http://deadbeef:2020
+      metricProvider: 
+        type: Prometheus
+        address: http://prometheus-k8s.monitoring.svc.cluster.local:9090
+  - name: LoadVariationRiskBalancing
+    args:
+      metricProvider:
+        type: Prometheus
+        address: http://prometheus-k8s.monitoring.svc.cluster.local:9090
+      safeVarianceMargin: 1.0
+      safeVarianceSensitivity: 1.0
+      watcherAddress: http://deadbeef:2020
+`),
+			wantProfiles: []defaultconfig.KubeSchedulerProfile{
+				{
+					SchedulerName: "scheduler-plugins",
+					PluginConfig: []defaultconfig.PluginConfig{
+						{
+							Name: coscheduling.Name,
+							Args: &config.CoschedulingArgs{
+								PermitWaitingTimeSeconds:      10,
+								DeniedPGExpirationTimeSeconds: 3,
+							},
+						},
+						{
+							Name: noderesources.AllocatableName,
+							Args: &config.NodeResourcesAllocatableArgs{
+								Mode: config.Least,
+								Resources: []v1.ResourceSpec{
+									{Name: string(corev1.ResourceCPU), Weight: 1000000},
+									{Name: string(corev1.ResourceMemory), Weight: 1},
+								},
+							},
+						},
+						{
+							Name: targetloadpacking.Name,
+							Args: &config.TargetLoadPackingArgs{
+								TargetUtilization: 60,
+								DefaultRequests: corev1.ResourceList{
+									corev1.ResourceCPU: testCPUQuantity,
+								},
+								DefaultRequestsMultiplier: "1.8",
+								WatcherAddress:            "http://deadbeef:2020",
+								MetricProvider: config.MetricProviderSpec{
+									Type:    config.Prometheus,
+									Address: "http://prometheus-k8s.monitoring.svc.cluster.local:9090",
+								},
+							},
+						},
+						{
+							Name: loadvariationriskbalancing.Name,
+							Args: &config.LoadVariationRiskBalancingArgs{
+								SafeVarianceMargin:      v1beta1.DefaultSafeVarianceMargin,
+								SafeVarianceSensitivity: v1beta1.DefaultSafeVarianceSensitivity,
+								WatcherAddress:          "http://deadbeef:2020",
+								MetricProvider: config.MetricProviderSpec{
+									Type:    config.Prometheus,
+									Address: "http://prometheus-k8s.monitoring.svc.cluster.local:9090",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
```

Comment:
Can you add more cases?

- leave some fields unspecified, and then verify they get defaulted properly
- provide some illegal args, and verify if we get the validation error

Reply:
done, thanks for the review!